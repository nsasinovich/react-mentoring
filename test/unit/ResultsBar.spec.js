import { shallow } from 'enzyme';
import React from 'react';
import ResultsBar from 'src/components/results_bar/results_bar';

describe('<ResultsBar />', () => {
    it('should render ResultsBar and match snapshot', () => {
        const wrapper = shallow(<ResultsBar />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should call onSortChange handler when sorting option is clicked', () => {
        const onClickCallback = jest.fn();
        const wrapper = shallow(<ResultsBar onSortChange={onClickCallback} />);
        expect(wrapper).toMatchSnapshot();

        const buttons = wrapper.find('.sorting-options li');
        buttons.forEach((btn) => {
            btn.simulate('click');
        });

        expect(onClickCallback).toHaveBeenCalledTimes(buttons.length);
    });

    it('should show results count message', () => {
        const count = 5;
        const wrapper = shallow(<ResultsBar count={count} />);
        expect(wrapper).toMatchSnapshot();

        expect(wrapper.find('.results-count').prop('children')).toBe(`${count} movies found`);
    });
});

