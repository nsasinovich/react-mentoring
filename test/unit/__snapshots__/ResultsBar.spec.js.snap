// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultsBar /> should call onSortChange handler when sorting option is clicked 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResultsBar
    count={0}
    onSortChange={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="results-count"
        >
          0 movies found
        </p>,
        <div
          className="results-sorting"
        >
          <p>
            Sort by
          </p>
          <ul
            className="sorting-options"
          >
            <li
              className=""
              onClick={[Function]}
            >
              <p>
                release date
              </p>
            </li>
            <li
              className=""
              onClick={[Function]}
            >
              <p>
                rating
              </p>
            </li>
          </ul>
        </div>,
      ],
      "className": "results-bar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "0 movies found",
          "className": "results-count",
        },
        "ref": null,
        "rendered": "0 movies found",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Sort by
            </p>,
            <ul
              className="sorting-options"
            >
              <li
                className=""
                onClick={[Function]}
              >
                <p>
                  release date
                </p>
              </li>
              <li
                className=""
                onClick={[Function]}
              >
                <p>
                  rating
                </p>
              </li>
            </ul>,
          ],
          "className": "results-sorting",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sort by",
            },
            "ref": null,
            "rendered": "Sort by",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li
                  className=""
                  onClick={[Function]}
                >
                  <p>
                    release date
                  </p>
                </li>,
                <li
                  className=""
                  onClick={[Function]}
                >
                  <p>
                    rating
                  </p>
                </li>,
              ],
              "className": "sorting-options",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "release date",
                "nodeType": "host",
                "props": Object {
                  "children": <p>
                    release date
                  </p>,
                  "className": "",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "release date",
                  },
                  "ref": null,
                  "rendered": "release date",
                  "type": "p",
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "rating",
                "nodeType": "host",
                "props": Object {
                  "children": <p>
                    rating
                  </p>,
                  "className": "",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "rating",
                  },
                  "ref": null,
                  "rendered": "rating",
                  "type": "p",
                },
                "type": "li",
              },
            ],
            "type": "ul",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="results-count"
          >
            0 movies found
          </p>,
          <div
            className="results-sorting"
          >
            <p>
              Sort by
            </p>
            <ul
              className="sorting-options"
            >
              <li
                className=""
                onClick={[Function]}
              >
                <p>
                  release date
                </p>
              </li>
              <li
                className=""
                onClick={[Function]}
              >
                <p>
                  rating
                </p>
              </li>
            </ul>
          </div>,
        ],
        "className": "results-bar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "0 movies found",
            "className": "results-count",
          },
          "ref": null,
          "rendered": "0 movies found",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Sort by
              </p>,
              <ul
                className="sorting-options"
              >
                <li
                  className=""
                  onClick={[Function]}
                >
                  <p>
                    release date
                  </p>
                </li>
                <li
                  className=""
                  onClick={[Function]}
                >
                  <p>
                    rating
                  </p>
                </li>
              </ul>,
            ],
            "className": "results-sorting",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sort by",
              },
              "ref": null,
              "rendered": "Sort by",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <li
                    className=""
                    onClick={[Function]}
                  >
                    <p>
                      release date
                    </p>
                  </li>,
                  <li
                    className=""
                    onClick={[Function]}
                  >
                    <p>
                      rating
                    </p>
                  </li>,
                ],
                "className": "sorting-options",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "release date",
                  "nodeType": "host",
                  "props": Object {
                    "children": <p>
                      release date
                    </p>,
                    "className": "",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "release date",
                    },
                    "ref": null,
                    "rendered": "release date",
                    "type": "p",
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": "rating",
                  "nodeType": "host",
                  "props": Object {
                    "children": <p>
                      rating
                    </p>,
                    "className": "",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "rating",
                    },
                    "ref": null,
                    "rendered": "rating",
                    "type": "p",
                  },
                  "type": "li",
                },
              ],
              "type": "ul",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ResultsBar /> should render ResultsBar and match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResultsBar
    count={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="results-count"
        >
          0 movies found
        </p>,
        <div
          className="results-sorting"
        >
          <p>
            Sort by
          </p>
          <ul
            className="sorting-options"
          >
            <li
              className=""
              onClick={[Function]}
            >
              <p>
                release date
              </p>
            </li>
            <li
              className=""
              onClick={[Function]}
            >
              <p>
                rating
              </p>
            </li>
          </ul>
        </div>,
      ],
      "className": "results-bar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "0 movies found",
          "className": "results-count",
        },
        "ref": null,
        "rendered": "0 movies found",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Sort by
            </p>,
            <ul
              className="sorting-options"
            >
              <li
                className=""
                onClick={[Function]}
              >
                <p>
                  release date
                </p>
              </li>
              <li
                className=""
                onClick={[Function]}
              >
                <p>
                  rating
                </p>
              </li>
            </ul>,
          ],
          "className": "results-sorting",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sort by",
            },
            "ref": null,
            "rendered": "Sort by",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li
                  className=""
                  onClick={[Function]}
                >
                  <p>
                    release date
                  </p>
                </li>,
                <li
                  className=""
                  onClick={[Function]}
                >
                  <p>
                    rating
                  </p>
                </li>,
              ],
              "className": "sorting-options",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "release date",
                "nodeType": "host",
                "props": Object {
                  "children": <p>
                    release date
                  </p>,
                  "className": "",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "release date",
                  },
                  "ref": null,
                  "rendered": "release date",
                  "type": "p",
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "rating",
                "nodeType": "host",
                "props": Object {
                  "children": <p>
                    rating
                  </p>,
                  "className": "",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "rating",
                  },
                  "ref": null,
                  "rendered": "rating",
                  "type": "p",
                },
                "type": "li",
              },
            ],
            "type": "ul",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="results-count"
          >
            0 movies found
          </p>,
          <div
            className="results-sorting"
          >
            <p>
              Sort by
            </p>
            <ul
              className="sorting-options"
            >
              <li
                className=""
                onClick={[Function]}
              >
                <p>
                  release date
                </p>
              </li>
              <li
                className=""
                onClick={[Function]}
              >
                <p>
                  rating
                </p>
              </li>
            </ul>
          </div>,
        ],
        "className": "results-bar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "0 movies found",
            "className": "results-count",
          },
          "ref": null,
          "rendered": "0 movies found",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Sort by
              </p>,
              <ul
                className="sorting-options"
              >
                <li
                  className=""
                  onClick={[Function]}
                >
                  <p>
                    release date
                  </p>
                </li>
                <li
                  className=""
                  onClick={[Function]}
                >
                  <p>
                    rating
                  </p>
                </li>
              </ul>,
            ],
            "className": "results-sorting",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sort by",
              },
              "ref": null,
              "rendered": "Sort by",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <li
                    className=""
                    onClick={[Function]}
                  >
                    <p>
                      release date
                    </p>
                  </li>,
                  <li
                    className=""
                    onClick={[Function]}
                  >
                    <p>
                      rating
                    </p>
                  </li>,
                ],
                "className": "sorting-options",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "release date",
                  "nodeType": "host",
                  "props": Object {
                    "children": <p>
                      release date
                    </p>,
                    "className": "",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "release date",
                    },
                    "ref": null,
                    "rendered": "release date",
                    "type": "p",
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": "rating",
                  "nodeType": "host",
                  "props": Object {
                    "children": <p>
                      rating
                    </p>,
                    "className": "",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "rating",
                    },
                    "ref": null,
                    "rendered": "rating",
                    "type": "p",
                  },
                  "type": "li",
                },
              ],
              "type": "ul",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ResultsBar /> should show results count message 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResultsBar
    count={5}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="results-count"
        >
          5 movies found
        </p>,
        <div
          className="results-sorting"
        >
          <p>
            Sort by
          </p>
          <ul
            className="sorting-options"
          >
            <li
              className=""
              onClick={[Function]}
            >
              <p>
                release date
              </p>
            </li>
            <li
              className=""
              onClick={[Function]}
            >
              <p>
                rating
              </p>
            </li>
          </ul>
        </div>,
      ],
      "className": "results-bar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "5 movies found",
          "className": "results-count",
        },
        "ref": null,
        "rendered": "5 movies found",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Sort by
            </p>,
            <ul
              className="sorting-options"
            >
              <li
                className=""
                onClick={[Function]}
              >
                <p>
                  release date
                </p>
              </li>
              <li
                className=""
                onClick={[Function]}
              >
                <p>
                  rating
                </p>
              </li>
            </ul>,
          ],
          "className": "results-sorting",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sort by",
            },
            "ref": null,
            "rendered": "Sort by",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li
                  className=""
                  onClick={[Function]}
                >
                  <p>
                    release date
                  </p>
                </li>,
                <li
                  className=""
                  onClick={[Function]}
                >
                  <p>
                    rating
                  </p>
                </li>,
              ],
              "className": "sorting-options",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "release date",
                "nodeType": "host",
                "props": Object {
                  "children": <p>
                    release date
                  </p>,
                  "className": "",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "release date",
                  },
                  "ref": null,
                  "rendered": "release date",
                  "type": "p",
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "rating",
                "nodeType": "host",
                "props": Object {
                  "children": <p>
                    rating
                  </p>,
                  "className": "",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "rating",
                  },
                  "ref": null,
                  "rendered": "rating",
                  "type": "p",
                },
                "type": "li",
              },
            ],
            "type": "ul",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="results-count"
          >
            5 movies found
          </p>,
          <div
            className="results-sorting"
          >
            <p>
              Sort by
            </p>
            <ul
              className="sorting-options"
            >
              <li
                className=""
                onClick={[Function]}
              >
                <p>
                  release date
                </p>
              </li>
              <li
                className=""
                onClick={[Function]}
              >
                <p>
                  rating
                </p>
              </li>
            </ul>
          </div>,
        ],
        "className": "results-bar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "5 movies found",
            "className": "results-count",
          },
          "ref": null,
          "rendered": "5 movies found",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Sort by
              </p>,
              <ul
                className="sorting-options"
              >
                <li
                  className=""
                  onClick={[Function]}
                >
                  <p>
                    release date
                  </p>
                </li>
                <li
                  className=""
                  onClick={[Function]}
                >
                  <p>
                    rating
                  </p>
                </li>
              </ul>,
            ],
            "className": "results-sorting",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sort by",
              },
              "ref": null,
              "rendered": "Sort by",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <li
                    className=""
                    onClick={[Function]}
                  >
                    <p>
                      release date
                    </p>
                  </li>,
                  <li
                    className=""
                    onClick={[Function]}
                  >
                    <p>
                      rating
                    </p>
                  </li>,
                ],
                "className": "sorting-options",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "release date",
                  "nodeType": "host",
                  "props": Object {
                    "children": <p>
                      release date
                    </p>,
                    "className": "",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "release date",
                    },
                    "ref": null,
                    "rendered": "release date",
                    "type": "p",
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": "rating",
                  "nodeType": "host",
                  "props": Object {
                    "children": <p>
                      rating
                    </p>,
                    "className": "",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "rating",
                    },
                    "ref": null,
                    "rendered": "rating",
                    "type": "p",
                  },
                  "type": "li",
                },
              ],
              "type": "ul",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
